File: May-LeetCoding-Challenge/05-First-Unique-Character-In-A-String/First-Unique-Character-In-A-String.java
class Solution {
    public int firstUniqChar(String s) {
        if(s == null || s.isEmpty()){
            return -1;
        }
        
        char[] num = new char[26];
        for(int i = 0; i < s.length(); i++) {
            num[s.charAt(i) - 'a']++;
        }
        
        for(int j = 0; j < s.length(); j++) {
            if(num[s.charAt(j) - 'a'] == 1)
                return j;
        }
        
        return -1;        
    }
}

File: May-LeetCoding-Challenge/05-First-Unique-Character-In-A-String/First-Unique-Character-In-A-String.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> num(26, 0);
        int ans = -1;
        for (auto &t: s) num[t - 'a']++;
        for (int i = 0; i < s.length(); i++)
            if (num[s[i] - 'a'] == 1) return i;
        return -1;
    }
};